packages/next/server/render.tsx

```diff
@@ -742,12 +742,27 @@ export async function renderToHTML(
   if (getServerSideProps && !isFallback) {
     let data: UnwrapPromise<ReturnType<GetServerSideProps>>
 
+    let canAccessRes = true
+    let resOrProxy = res
+    if (process.env.NODE_ENV !== 'production') {
+      resOrProxy = new Proxy(res, {
+        get: function (obj: ServerResponse, prop: string, receiver: any) {
```

Comment:
Out of curiosity, if we type this as `ServerResponse | Proxy<ServerResponse>`, do we still need to type the `get` function here or will it be inferred?

Reply:
It will be inferred if we add a generic to the Proxy like `new Proxy<ServerResponse>`. Just changed!